### ISSUE TITLE
handling api version on request

### ISSUE THREAD

**Issue by m-matth** [adamflott/milena#8]
Hi !

i'm trying to manage multiple version of api in milena. Could be nice to have, because sometime it has a big impact, like OffsetCommit that can use zookeeper or kafka for storage depending on version used.

i'm testing to store the version in kafkaState, not sure if it the good aproach through :

https://github.com/matth-/milena/commit/cdf3a49ba727b43d38b27de12c426851958482a1

i still had to find a way to constrain request/response data type to its own version.

**Comment by adamflott** [comment]
Yeah, now that I'm reinvesting my time into getting this library up to the latest protocol, I haven't thought of a good way to handle version(s) yet. 

Your change does look like a move in the right direction. I wonder if it's worth investing in how to handle every version without duplicating code. I should look into how other popular implementations of their protocol handle this.

**Comment by m-matth** [comment]
you're right, code reduction can be left in a refactoring process with a clear view of protocol (and more skilled haskeller than i :).

will try to find time to finish the work on it, feel free to reuse it if you progress on the topic.

cheers
